src/integrationTest/java/uk/gov/hmcts/dts/fact/repositories/CourtPostcodeRepositoryTest.java

```diff
@@ -0,0 +1,34 @@
+package uk.gov.hmcts.dts.fact.repositories;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import uk.gov.hmcts.dts.fact.entity.Court;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
+@DataJpaTest
+public class CourtPostcodeRepositoryTest {
+    private static final String TEST_MANCHESTER_COURT_SLUG = "manchester-civil-justice-centre-civil-and-family-courts";
+    private static final String TEST_MANCHESTER_POSTCODE = "M11";
+    private static final String TEST_BIRMINGHAM_POSTCODE = "B11";
+
+    @Autowired
+    private CourtPostcodeRepository courtPostcodeRepository;
+
+    @Autowired
+    private CourtRepository courtRepository;
+
+    @Test
+    void shouldRetrieveCourtPostcodeUsingAValidPostcodeOnly() {
```

Comment:
Is it worth doing an integration test for add and delete as well? Based on the new routes that have been added 

Might be complex but doable with a setup/teardown? Could be worth us having a chat

Reply:
These will be done in the functional tests by ashwini